OpenRA.Mods.Common/Widgets/VideoPlayerWidget.cs

```diff
@@ -257,33 +259,38 @@ public void PlayThen(Action after)
 				return;
 
 			onComplete = after;
-			if (stopped && Video.HasAudio)
+			if (startTime == null && Video.HasAudio)
 				Game.Sound.PlayVideo(Video.AudioData, Video.AudioChannels, Video.SampleBits, Video.SampleRate);
 			else
 				Game.Sound.PlayVideo();
 
-			stopped = Paused = false;
+			startTime ??= DateTime.Now;
+
+			if (pauseTime != null)
+				startTime += DateTime.Now - pauseTime;
```

Comment:
is this needed? seems a big odd, adjust the start time. i think it would be more clear to maintain a totalTimePaused variable and subtract it where needed.

Reply:
IMO the concept of tracking pause time is odd. It's completely unnecessary and just adds unneeded complexity. It would be better to use a `StopWatch` from `System.Diagnostics`